File: DSA/DivyanshKumarSingh_590012175/Day_3_Question_2.py
def duplicate(lst):
    '''
    Method 1:
    for i in range(0, size):
        c=0
        for j in range(i+1, size):
            if (lst[i]==lst[j]):
                return True
    return False
    
    Method 2:
    return (len(lst) != len(set(lst)))

    '''
    #Method 3: (Set-Based Thinking)
    seen = set()
    for num in lst:
        if num in seen:
            return True
        seen.add(num)
    return False


size = int(input("Enter size of Array: "))
print(f"Enter {size} elements: ")

lst = []
for i in range(size):
    element = int(input(f"Enter element {i+1}: "))
    lst.append(element)

print (duplicate(lst))

File: DSA/AYUSHMEHTA_590015131/day3question2.cpp
#include<iostream>
#include<vector>
#include <unordered_set>
using namespace std;

bool duplicate(vector<int>&arr,int n){
        unordered_set<int> check;
         for (int num : arr) {
        if (check.count(num)) return true;
        check.insert(num);
    }
    return false; 

}




int main(){
    int n;
    cout<<"Enter the no of element for array:"<<" ";
    cin>>n;
    vector<int>arr(n);
    cout<<"enter the elements for the array:"<< " ";
    for(int i=0;i<n;i++) cin>>arr[i];
    
  bool duplicates = duplicate(arr, n);

    if (duplicates)
        cout << "Duplicates found in the array." << endl;
    else
        cout << "No duplicates found in the array." << endl;

    return 0;

}